packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -306,6 +321,33 @@ EmbedFlutterFrameworks() {
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/App.framework/App"
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/Flutter.framework/Flutter"
   fi
+
+  AddObservatoryBonjourService
+}
+
+# Add the observatory publisher Bonjour service to the produced app bundle Info.plist.
+AddObservatoryBonjourService() {
+  ParseFlutterBuildMode
+  local build_mode="${retval}"
+  # Debug and profile only.
+  if [[ "${build_mode}" == "release" ]]; then
+    return
+  fi
+  local built_products_plist="${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}"
```

Comment:
I might be misreading this a bit, isn't this ultimately the same plist as https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_system/targets/ios.dart#L341 ?

Reply:
It's not the same file, that's the project's `ios/Runner/Info.plist` (build setting `INFOPLIST_FILE`), and that's checked into source code, and where [the docs](https://flutter.dev/docs/deployment/ios#updating-the-apps-name) tell users to update some fields before they submit to the app store.  

This PR is editing the Xcode-generated Info.plist that gets embedded into the `Runner.app`.
After Xcode compiles everything and it embeds resources (storyboards, images, etc), it creates a `Runner.app/Info.plist`, which is a combination what's in the user's Info.plist in Xcode, plus a bunch of other Xcode-generated keys and metadata
<img width="1032" alt="Screen Shot 2020-09-01 at 12 05 55 AM" src="https://user-images.githubusercontent.com/682784/91813044-eb3efa80-ebe6-11ea-93da-b3033cf9622b.png">

By editing this final generated Info.plist, we don't need to update the user's checked in Info.plist, update the template, or have a complicated migration where we set different Info.plists depending on configuration, which would cause flavor-users a bunch of pain.  We know the user needs this service in debug and profile when copied to the device, and it also means it won't slip into the release version that gets submitted to the App Store.

Reply:
Actually, looking again, https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_system/targets/ios.dart#L341 is a different third `Info.plist`. It's the one that gets embedded in the `App.framework`.

Reply:
Does it matter which Plist these permissions are added to, or could it be done to the flutter framework plist itself?

Reply:
It needs to be on the app bundle, so `Runner.app/Info.plist`.  A framework can't set app-level permission requests or entitlements.  Otherwise this problem would be a whole lot simpler.

Reply:
understood.